release/serve_tests/workloads/serve_resnet_benchmark.py

```diff
@@ -0,0 +1,167 @@
+        ) as response:
+            return await response.json()
+
+    async with aiohttp.ClientSession() as session:
+        for _ in range(requests_sent):
+            start = time.time()
+            res = await fetch(session)
+            end = time.time()
+            tps_stats.append(data_size / (end - start))
+            model_inference_stats.append(res["model_inference_latency"])
+
+    return tps_stats, model_inference_stats
+
+
+async def trial(measure_func, data_size: int = 8, num_clients: int = 1):
+
+    client_tasks = [measure_func for _ in range(num_clients)]
+
+    result_stats_list = await asyncio.gather(
+        *[client_task(data_size) for client_task in client_tasks]
+    )
+
+    throughput_stats_tps = []
+    for client_stats in result_stats_list:
+        throughput_stats_tps.extend(client_stats[0])
+    throuput_mean = round(np.mean(throughput_stats_tps), 2)
+
+    model_inference_latency = []
+    for client_stats in result_stats_list:
+        model_inference_latency.extend(client_stats[1])
+    inference_latency_mean = round(np.mean(model_inference_latency), 2)
+
+    return throuput_mean, inference_latency_mean
+
+
+async def json_resolver(request: starlette.requests.Request):
+    return await request.json()
+
+
+@click.command()
+@click.option("--gpu-env", type=bool, is_flag=True, default=False)
+@click.option("--smoke-run", type=bool, is_flag=True, default=False)
+def main(gpu_env: Optional[bool], smoke_run: Optional[bool]):
+
+    loop = asyncio.get_event_loop()
+    test_name = "resnet50_cpu"
+    device = "cpu"
+    if gpu_env:
+        test_name = "resnet50_gpu"
+        device = "cuda"
+        ImageObjectioner.set_options(ray_actor_options={"num_gpus": 1})
+
+    # batch size
+    batch_sizes = [16, 32, 64]
+
+    with InputNode() as user_input:
+        io = ImageObjectioner.bind(DataDownloader.bind(), device=device)
+        dag = DAGDriver.bind(io.predict.bind(user_input), http_adapter=json_resolver)
+        handle = serve.run(dag)
+
+    if smoke_run:
+        res = handle.predict.remote(input_uris)
+        print(ray.get(res))
+
+    else:
+        result = {}
+        print("warming up...")
+        for _ in range(10):
+            res = handle.predict.remote([input_uris[0]])
+        print("start load testing...")
+        for batch_size in batch_sizes:
+            throughput_mean_tps, model_inference_latency_mean = loop.run_until_complete(
+                trial(measure_http_throughput_tps, batch_size)
+            )
+            result[f"batch size {batch_size}"] = {
+                "throughput_mean_tps": throughput_mean_tps,
+                "model_inference_latency_mean": model_inference_latency_mean,
+            }
+            print(throughput_mean_tps, model_inference_latency_mean)
+
+        save_test_results(
+            {test_name: result},
+            default_output_file="/tmp/serve_resent_benchmark.json",
```

Comment:
Do you know how the release test infra finds this file? It might have to be named `/tmp/release_test_out.json` or use the env var `TEST_OUTPUT_JSON` in order for the "fetch results" step to work, do you mind double checking this?

Reply:
+1 please use `TEST_OUTPUT_JSON` and follow other files for the JSON schema.

Reply:
so it can be shown in our perf dashboard